front/app/containers/Admin/projects/project/events/edit.tsx

```diff
@@ -324,23 +320,26 @@ const AdminProjectEventEdit = () => {
               new Date(newAttributes['start_at']),
               duration
             ).toISOString();
-          } else if (
-            name === 'end_at' &&
-            newAttributes['end_at'] &&
-            isStartDateAfterEndDate
-          ) {
-            const duration = moment
-              .duration(
-                moment(previousState['end_at']).diff(
-                  moment(newAttributes['start_at'])
+          } else if (name === 'end_at' && newAttributes['end_at']) {
+            const isStartDateAfterEndDate =
```

Comment:
Only relevant when this part of the if block is triggered, so moved it here.

Reply:
Agreed it's not ideal. Best imo would be to have two separate functions for start_at and end_at instead of the if block.